src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,236 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.util.DateTimeHandler;
+
+/**
+ * Encapsulates the handling of user's input.
+ */
+public class Parser {
+
+    /** Storage for parser to write data to. */
+    private final Storage storage;
+
+    /** UI for parser to display messages. */
+    private final Ui ui;
+
+    /** Task list for parser to add/edit/delete tasks. */
+    private final TaskList taskList;
+
+    /**
+     * Creates a new parser from a storage.
+     *
+     * @param storage Storage of Duke's data.
+     */
+    public Parser(Storage storage) {
+        this.storage = storage;
+        this.ui = this.storage.getUi();
+        this.taskList = this.storage.getTaskList();
+    }
+
+    /**
+     * Processes the input fed to Duke.
+     *
+     * @param input Input string.
+     * @throws DukeException If input is invalid.
+     */
+    public void processInput(String input) throws DukeException {
+        if (input.equals("bye")) {
+            this.ui.printMessage("Bye. Hope to see you again soon!");
+        } else if (input.equals("list")) {
+            this.ui.printTaskList(this.taskList);
+        } else if (input.contains("delete")) {
+            deleteTask(input);
+        } else if (input.contains("done")) {
+            markTaskAsDone(input);
+        } else if (input.contains("get")) {
+            printTasksFromDate(input);
+        } else if (input.contains("find")) {
+            findTasks(input);
+        } else if (input.contains("todo") || input.contains("event") || input.contains("deadline")) {
+            createTask(input);
+        } else {
+            throw new DukeException("I'm sorry, but I don't know what that means. \u2639\n");
+        }
+    }
+
+    /**
+     * Returns a footer containing the number of tasks in the list.
+     *
+     * @return Footer string containing the number of tasks.
+     */
+    private String numOfTasksFooter() {
```

Comment:
Perhaps can consider changing the method name into getNumOfTasks().

Reply:
Changed to getNumOfTaskFooter() to get the footer message when printing out messages.